### ISSUE TITLE
Bug: sort-classes should consider initialization order

### ISSUE THREAD

**Issue by danvk** [azat-io/eslint-plugin-perfectionist#102]
### Describe the bug

If a class's properties include initializers, then perfectionist/sort-classes must consider references between those properties when sorting them.

Sorting property `a` before property `z` is incorrect if `z` references `a` in its initializer (see code sample below).

### Code example

perfectionist/sort-classes reports an error on this class:

```ts
class C {
  z = new Date();
  a = this.z;
}
// Expected "a" to come before "z"
```

Autofixing the error "corrects" the class to this:

```ts
class C {
  a = this.z;
  //       ~ Property 'z' is used before its initialization.ts(2729)
  z = new Date();
}
```

But this has a type error and `a` [will be undefined at runtime][1].

[1]: https://www.typescriptlang.org/play?#code/MYGwhgzhAEDC0G8BQ1pmgXmgFwBYEsIA6ALwG4VoB6K1O1AP2gAUAnAewAcBTV7AT2gByEkOiFoAVwjcAJtABG3AGbtW3cdhj4Advmz4wIfCTAH2OoloAUAJgDstgJwBKSiUzQd3AO7QAImbc1i4UAL5ISMAWENjQwJ7efrAhFNE6EOwg3EQg7ADm1sBEYKFIQA

### ESLint version

v8.56.0

### ESLint Plugin Perfectionist version

v2.5.0

### Additional comments

What's the correct behavior here? Ideal behavior would be to sort as usual while still preserving the initialization graph. Maybe slightly easier would be to preserve the relative order of any properties that reference another one in their initializer, or are referenced in an initializer.

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by Tragio** [comment]
I'm facing a similar issue in our codebase. @azat-io do you have any fix in the works? 😄 

Thank you for your amazing work!! 🙏

**Comment by azat-io** [comment]
Thank you for your issue!

This bug was fixed in 8c35a7dfbe9b841f86f303c4e9fd4170b47c9c0b and released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054

**Comment by niemyjski** [comment]
I'm running into this on 3.0 where fix is putting this before, then then lint check is failing

```

        <svelte:component
            this={icon}
            class="h-6 w-6 text-muted-foreground transition duration-75 group-hover:text-foreground {active ? 'text-foreground' : ''}"
        />
        ```
        
        
        \ClientApp\src\routes\(app)\(components)\layouts\SidebarMenuItem.svelte
  30:13  error  Expected "class" to come before "this"  perfectionist/sort-svelte-attributes
